actors/builtin/miner/miner_actor.go

```diff
@@ -1701,13 +1716,14 @@ func handleProvingDeadline(rt Runtime) {
 			pledgeDeltaTotal = big.Add(pledgeDeltaTotal, result.PledgeDelta)
 
 			penaltyTarget := big.Add(declaredPenalty, undeclaredPenalty)
-			if !penaltyTarget.IsZero() {
-				unlockedBalance := st.GetUnlockedBalance(rt.CurrentBalance())
-				penaltyFromVesting, penaltyFromBalance, err := st.PenalizeFundsInPriorityOrder(store, currEpoch, penaltyTarget, unlockedBalance)
-				builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to unlock penalty")
-				penaltyTotal = big.Sum(penaltyTotal, penaltyFromVesting, penaltyFromBalance)
-				pledgeDeltaTotal = big.Sub(pledgeDeltaTotal, penaltyFromVesting)
-			}
+
+			err = st.ApplyPenalty(penaltyTarget)
+			builtin.RequireNoErr(rt, err, exitcode.Unwrap(err, exitcode.ErrIllegalState), "failed to apply penalty")
+
+			penaltyFromVesting, penaltyFromBalance, err := st.RepayPartialDebtInPriorityOrder(store, currEpoch, rt.CurrentBalance())
+			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to unlock penalty")
+			penaltyTotal = big.Add(penaltyFromVesting, penaltyFromBalance)
```

Comment:
The reason this is `big.Add` is because (IMO) `penaltyTotal` is misnamed - it's actually the amount to burn, rather than the total amount penalized. See a similar misnomer above, where `depositToBurn` is never explicitly burned, but rather added to `FeeDebt`

Reply:
As @wadeAlexC is getting at I explicitly changed from Sum to Add for clarity.  This should be the only place funds to burn are incremented.  The design from #1026 means that calls to `ApplyPenalty` will capture any additional penalizations in the state's fee debt.

I can change the name from penaltyTotal to penaltyToBurn